core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorAdapterTest.scala

```diff
@@ -135,4 +134,48 @@ class GroupCoordinatorAdapterTest {
     assertEquals(expectedData, future.get())
   }
 
+  @Test
+  def testListGroups(): Unit = {
+    testListGroups(null, Set.empty)
+    testListGroups(List(), Set.empty)
+    testListGroups(List("Stable"), Set("Stable"))
+  }
+
+  def testListGroups(
+    statesFilter: List[String],
+    expectedStatesFilter: Set[String]
+  ): Unit = {
+    val groupCoordinator = mock(classOf[GroupCoordinator])
+    val adapter = new GroupCoordinatorAdapter(groupCoordinator)
+
+    val ctx = makeContext(ApiKeys.LIST_GROUPS.latestVersion)
+    val data = new ListGroupsRequestData()
+      .setStatesFilter(statesFilter.asJava)
+
+    when(groupCoordinator.handleListGroups(expectedStatesFilter)).thenReturn {
+      (Errors.NOT_COORDINATOR, List(
+        GroupOverview("group1", "protocol1", "Stable"),
+        GroupOverview("group2", "qwerty", "Empty")
```

Comment:
Should we be including the empty group if the filter is stable?

Reply:
I agree that this is a tad weird. However, the filtering is not done by `GroupCoordinatorAdapter` so the test only verifies that inputs/outputs are correctly translated here. In practice, we would not have an error and a list of groups as well.

Reply:
Hmm I think I'm a bit confused by your last sentence. 
But to clarify my understanding -- this filtering is done within the handleListGroups method, but since this test is mocking the result we don't actually filter the groups? 

And the different filters at the top are just really checking the empty state filter is tested correctly?

Reply:
That’s right.